buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java

```diff
@@ -348,10 +352,22 @@ private String runForbiddenAPIsCli() throws IOException {
             }
             spec.setIgnoreExitValue(true);
         });
+        if (OS.current().equals(OS.LINUX) && result.getExitValue() == SIG_KILL_EXIT_VALUE) {
+            throw new IllegalStateException(
+                "Third party audit was killed buy SIGKILL, could be a victim of the Linux OOM killer"
+            );
+        }
         final String forbiddenApisOutput;
         try (ByteArrayOutputStream outputStream = errorOut) {
             forbiddenApisOutput = outputStream.toString(StandardCharsets.UTF_8.name());
         }
+        if (Arrays.asList(
+            CliMain.EXIT_SUCCESS,
+            CliMain.EXIT_VIOLATION,
+            CliMain.EXIT_UNSUPPORTED_JDK
+        ).contains(result.getExitValue()) == false) {
+            throw new IllegalStateException("Forbidden APIs cli failed: " + forbiddenApisOutput);
```

Comment:
We should ensure we dump the output to the console on this condition as well.

Reply:
We do add it to the exception here, do you mean the other one when the process is killed ? 
I don't think that's relevant it will just be something saying it's starting up  etc from what  I saw.

Reply:
Sorry, I didn't catch that. That seems fine.